src/bincode/bincode.zig

```diff
@@ -592,8 +671,14 @@ pub fn FieldConfig(comptime T: type) type {
         serializer: ?SerializeFunction = null,
         free: ?FreeFunction = null,
         skip: bool = false,
-        default_on_eof: bool = false,
-        default_fn: ?fn (alloc: std.mem.Allocator) T = null,
+        hashmap: if (hashMapInfo(T)) |hm_info| bincode.HashMapConfig(hm_info) else void = if (hashMapInfo(T) != null) .{} else {},
```

Comment:
This feels oddly specific. Do you foresee adding other fields here for other commonly used types?

My instinct is that HashMap and ArrayList should just use FieldConfig like any other struct that requires custom logic.

The problem with that is you need to specify a field config everywhere you use these types, which is annoying. A compromise would be that they're defined within bincode and used by default for those types, but they are still just a normal FieldConfig, and the custom logic outside FieldConfig is minimal.

I don't expect you to rewrite anything in this PR, just curious about your thoughts on this topic.

Reply:
I have a broader plan to extract this out of FieldConfig as well, which involves actually generalizing `FieldConfig` into `Config`, and providing this functionality through a `hash_map` namespace.
So my thoughts are: yeah this is icky, but it works and helps me further my goal of bincode simplification. Just out of scope for this PR.